plugins/modules/aci_tenant_action_rule_profile.py

```diff
@@ -249,18 +405,72 @@ def main():
             module_object=action_rule,
             target_filter={"name": action_rule},
         ),
+        child_classes=list(child_classes.keys()),
     )
 
     aci.get_existing()
 
     if state == "present":
+        child_configs = []
+        for class_name, attribute in child_classes.items():
+            attribute_input = attribute.get("attribute_input")
+            # This condition enables to user to keep its previous configurations if they are not passing anyting in the payload.
+            if attribute_input is not None:
+                # This condition checks if the attribute input is a dict and checks if all of its values are None (stored as a boolean in only_none).
+                only_none = False
+                if isinstance(attribute_input, dict):
+                    only_none = all(value is None for value in attribute_input.values())
+                # This condition checks if the child object needs to be deleted depending on the type of the corresponding attribute input (bool, str, dict).
+                if attribute_input == "" or attribute_input is False or only_none:
+                    if isinstance(aci.existing, list) and len(aci.existing) > 0:
```

Comment:
conditional can be combined
```suggestion
                if (attribute_input == "" or attribute_input is False or only_none) and isinstance(aci.existing, list) and len(aci.existing) > 0:
```